### ISSUE TITLE
Cannot import CSV

### ISSUE THREAD

**Issue by DemokritosTheAtomist** [jameshnsears/QuoteUnquote#174]
Dear James,

First of all, thank you for your work and this app !
My issue is that my CSV file doesn't work when I try to import it.
I exported your r/quote/top CSV and replaced your Source and Quotes by mine but when I try to import it this message appears : 
"Invalid .csv format: Index for header 'Quotation' is 1 but CSVRecord only has..." and I can't see the end of the message.

I'm not a computer scientist but I understand most of the basics, now I'm lost haha !

Thank you by advance !

**Comment by jameshnsears** [comment]
Hi - can you upload you .csv file to this issue? Alternatively, if you prefer, email it to me directly - james.hn.sears@gmail.com -and I will investigate.

Thanks for raising this issue by the way - I do appreciate it.

**Comment by jameshnsears** [comment]
NOTE to self: possible solituion (?) = 
https://m3.material.io/components/snackbar/overview
or
AlertDialog.Builder builder = new AlertDialog.Builder(context);
builder.setTitle("Your title");
builder.setMessage("Your longer text here");
builder.setPositiveButton("OK", null);
builder.show();

Area: QuotationsDatabaseFragment.setHandleImportCsv

**Comment by DemokritosTheAtomist** [comment]
Just for the follow-up I sent you an email with the CSV, and here is a screenshot of the error message.

![Screenshot_20230704-175009.png](https://github.com/jameshnsears/QuoteUnquote/assets/138607501/74f957d7-3f1e-4499-90ed-4b4bf90b8efe)

**Comment by jameshnsears** [comment]
Hi - thanks for both the private email, with the .csv, and the above screenshot. Both are helpful.

Firstly, the.csv is "wrong" (if I can be so blunt) in that some of the lines look like this:
**author||quotation 1
2
3**

The .csv processing software I use - Apache Commons CSV - I have coded to only accept lines that looks like this:
**author||quotation 1 2 3**

It's possible for Apache Commons CSV to process multiple lines for a quotation, but there are some unpleasant side effects.

---
Another issue is that some of the .csv lines look like this:
**"author||quotation"
author||quotation**

Ideally, all lines should look like:
**author||quotation**

---
The final issue, as I see it, is that the software needs to be more descriptive when importing a .csv file. I'll change the code so that it shows a longer error message (a Snackbar rather than a Toast) and I'll also include a line # in the message.

The next release, hopefully published within a few days, will contain this change.

I won't close this issue until I've published a release. So if you have anything to add / suggest then please update this issue or contact me directly.

Thank you for taking the time to contact me and hep improve the app!

**Comment by DemokritosTheAtomist** [comment]
Thank you for your time and all these details.

I don't know how my file appears on your PC but on mine it is like that 
![image](https://github.com/jameshnsears/QuoteUnquote/assets/138607501/d61c209d-9e21-44dc-bdd1-cd11e8d60415)
author 1 ||quote 1
author 1 ||quote 2
author 1 ||quote 3
Each line is one quote, not a huge one.

I am not sure to understand the format author||quotation 1 2 3,. You are telling me that I can't put the same author on more than 1 line ?

And I don't understand the issue
"author||quotation"
author||quotation

Could you upload an excel file as an example ? 

Thank you very much, and have a great day !

**Comment by DemokritosTheAtomist** [comment]
I will let my previous comment just because I am interested in your remarkes concerning my points.
I found the main issue, it apparently was the fact that as you said, somme of the quote were on several lines (but in the same cell). I replaced the line break with simple spaces and now it works.
Last thing, is that the accents appears as diamond with a question mark, but I can easily clear the issue on my CSV.

Thank you for your help and great job anyway !!!
